### ISSUE TITLE
How can I recover the covariance matrix using quaternions and two-dimensional scales?

### ISSUE THREAD

**Issue by SlamCabbage** [hbb1/2d-gaussian-splatting#88]

**Comment by hbb1** [comment]
padding zero to the scales and pass them into `build_scaling_rotation`

**Comment by SlamCabbage** [comment]
torch.exp is used in build_scaling_rotation, so is the "scales" saved at the end the actual size of the Gaussian in this scene?

**Comment by hbb1** [comment]
Hi, when I refers to the scales, I mean the post-activation scales.  Yes, the scales are indeed measured in the actual scene; that is, the world space. You need to append the scale-post-activation for recovering a full covariance.

**Comment by SlamCabbage** [comment]
![image](https://github.com/hbb1/2d-gaussian-splatting/assets/95751923/e9539715-819e-4bbc-8c74-bac42da8d8a6)
大佬您好，我用中文了，这是waymo数据，我可视化它的协方差，这个是按照scale的0.05倍可视化的。如果按照1.0倍进行可视化，那么整个场景都是一坨纠缠在一起的，完全看不出来。

**Comment by DavidXu-JJ** [comment]
Waymo should not be a problem.
https://streetunveiler.github.io/
I use 2DGS in my project above. And I at least successfully reconstruct the scene.

**Comment by SlamCabbage** [comment]
![image](https://github.com/hbb1/2d-gaussian-splatting/assets/95751923/caf4f155-baee-4c9a-b283-90438b2f3020)
我对scales做了一个exp操作之后，可视化的结果如上图。
用2d gs可视化工具能看到这个场景的地面点虽然不多，但是在图像上是覆盖全了，为什么我对高斯进行可视化，地面并没有能够把空出补全的大高斯平面呢？

**Comment by SlamCabbage** [comment]
> Waymo should not be a problem. https://streetunveiler.github.io/ I use 2DGS in my project above. And I at least successfully reconstruct the scene.

Already following, looking forward to open source
